### ISSUE TITLE
avg_align_error not changing for Glow TTS [Bug] 

### ISSUE THREAD

**Issue by lihaoyangML** [coqui-ai/TTS#652]
Hi sir, I think [issue 408](https://github.com/coqui-ai/TTS/issues/408) is still present in both v0.0.12 and v0.013. When I used [glow_tts_ljspeech.json](https://github.com/coqui-ai/TTS/blob/v0.0.13/TTS/tts/configs/glow_tts_ljspeech.json) to run a training on LJSpeech, the avg_align_loss remained constant. Here is the Tensorboard output:

![image](https://user-images.githubusercontent.com/60748402/125248325-81bf6a80-e326-11eb-8373-23965fd581c7.png)

**To Reproduce**

#### Installation of the library
git clone https://github.com/coqui-ai/TTS.git
git reset --hard f02f033
pip3 install -e .
pip3 install packaging
pip3 install numba==0.48
pip3 install umap-learn==0.4.6

#### updating the path information in glow_tts_ljspeech.json
{
        "model": "glow_tts",
        "run_name": "glow-tts-residual_bn_conv",
        "run_description": "glow-tts model training with residual BN conv.",

        // AUDIO PARAMETERS
        "audio":{
            "fft_size": 1024,         // number of stft frequency levels. Size of the linear spectogram frame.
            "win_length": 1024,      // stft window length in ms.
            "hop_length": 256,       // stft window hop-lengh in ms.
            "frame_length_ms": null, // stft window length in ms.If null, 'win_length' is used.
            "frame_shift_ms": null,  // stft window hop-lengh in ms. If null, 'hop_length' is used.

            // Audio processing parameters
            "sample_rate": 22050,   // DATASET-RELATED: wav sample-rate. If different than the original data, it is resampled.
            "preemphasis": 0.0,     // pre-emphasis to reduce spec noise and make it more structured. If 0.0, no -pre-emphasis.
            "ref_level_db": 0,     // reference level db, theoretically 20db is the sound of air.

            // Griffin-Lim
            "power": 1.1,           // value to sharpen wav signals after GL algorithm.
            "griffin_lim_iters": 60,// #griffin-lim iterations. 30-60 is a good range. Larger the value, slower the generation.

            // Silence trimming
            "do_trim_silence": true,// enable trimming of slience of audio as you load it. LJspeech (false), TWEB (false), Nancy (true)
            "trim_db": 60,          // threshold for timming silence. Set this according to your dataset.

            // MelSpectrogram parameters
            "num_mels": 80,         // size of the mel spec frame.
            "mel_fmin": 50.0,        // minimum freq level for mel-spec. ~50 for male and ~95 for female voices. Tune for dataset!!
            "mel_fmax": 7600.0,     // maximum freq level for mel-spec. Tune for dataset!!
            "spec_gain": 1.0,         // scaler value appplied after log transform of spectrogram.00

            // Normalization parameters
            "signal_norm": false,   // normalize spec values. Mean-Var normalization if 'stats_path' is defined otherwise range normalization defined by the other params.
            "min_level_db": -100,   // lower bound for normalization
            "symmetric_norm": true, // move normalization to range [-1, 1]
            "max_norm": 1.0,        // scale normalization to range [-max_norm, max_norm] or [0, max_norm]
            "clip_norm": true,      // clip normalized values into the range.
            "stats_path": null    // DO NOT USE WITH MULTI_SPEAKER MODEL. scaler stats file computed by 'compute_statistics.py'. If it is defined, mean-std based notmalization is used and other normalization params are ignored
        },

        // VOCABULARY PARAMETERS
        // if custom character set is not defined,
        // default set in symbols.py is used
        // "characters":{
        //     "pad": "_",
        //     "eos": "~",
        //     "bos": "^",
        //     "characters": "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz!'(),-.:;? ",
        //     "punctuations":"!'(),-.:;? ",
        //     "phonemes":"iyɨʉɯuɪʏʊeøɘəɵɤoɛœɜɞʌɔæɐaɶɑɒᵻʘɓǀɗǃʄǂɠǁʛpbtdʈɖcɟkɡqɢʔɴŋɲɳnɱmʙrʀⱱɾɽɸβfvθðszʃʒʂʐçʝxɣχʁħʕhɦɬɮʋɹɻjɰlɭʎʟˈˌːˑʍwɥʜʢʡɕʑɺɧɚ˞ɫ"
        // },

        "add_blank": false, // if true add a new token after each token of the sentence. This increases the size of the input sequence, but has considerably improved the prosody of the GlowTTS model.

        // DISTRIBUTED TRAINING
        "distributed":{
            "backend": "nccl",
            "url": "tcp:\/\/localhost:54321"
        },

        "reinit_layers": [],    // give a list of layer names to restore from the given checkpoint. If not defined, it reloads all heuristically matching layers.

        // MODEL PARAMETERS
        // "use_mas": false,       // use Monotonic Alignment Search if true. Otherwise use pre-computed attention alignments.
        "hidden_channels_encoder": 192,
        "hidden_channels_decoder": 192,
        "hidden_channels_duration_predictor": 256,
        "use_encoder_prenet": true,
        "encoder_type": "rel_pos_transformer",
        "encoder_params": {
            "kernel_size":3,
            "dropout_p": 0.1,
            "num_layers": 6,
            "num_heads": 2,
            "hidden_channels_ffn": 768,
            "input_length": null
        },

        // TRAINING
        "batch_size": 32,       // Batch size for training. Lower values than 32 might cause hard to learn attention. It is overwritten by 'gradual_training'.
        "eval_batch_size":16,
        "r": 1,                 // Number of decoder frames to predict per iteration. Set the initial values if gradual training is enabled.
        "loss_masking": true,   // enable / disable loss masking against the sequence padding.
        "mixed_precision": true,
        "data_dep_init_iter": 10,

        // VALIDATION
        "run_eval": true,
        "test_delay_epochs": 0,       //Until attention is aligned, testing only wastes computation time.
        "test_sentences_file": null,  // set a file to load sentences to be used for testing. If it is null then we use default english sentences.

        // OPTIMIZER
        "noam_schedule": true,         // use noam warmup and lr schedule.
        "grad_clip": 5.0,              // upper limit for gradients for clipping.
        "epochs": 10000,               // total number of epochs to train.
        "lr": 1e-3,                    // Initial learning rate. If Noam decay is active, maximum learning rate.
        "wd": 0.000001,                // Weight decay weight.
        "warmup_steps": 4000,          // Noam decay steps to increase the learning rate from 0 to "lr"
        "seq_len_norm": false,         // Normalize eash sample loss with its length to alleviate imbalanced datasets. Use it if your dataset is small or has skewed distribution of sequence lengths.

        // TENSORBOARD and LOGGING
        "print_step": 25,       // Number of steps to log training on console.
        "tb_plot_step": 100,    // Number of steps to plot TB training figures.
        "print_eval": false,     // If True, it prints intermediate loss values in evalulation.
        "save_step": 5000,      // Number of training steps expected to save traninpg stats and checkpoints.
        "checkpoint": true,     // If true, it saves checkpoints per "save_step"
        "keep_all_best": false,  // If true, keeps all best_models after keep_after steps
        "keep_after": 10000,    // Global step after which to keep best models if keep_all_best is true
        "tb_model_param_stats": false,     // true, plots param stats per layer on tensorboard. Might be memory consuming, but good for debugging.

        // DATA LOADING
        "text_cleaner": "phoneme_cleaners",
        "enable_eos_bos_chars": false, // enable/disable beginning of sentence and end of sentence chars.
        "num_loader_workers": 4,        // number of training data loader processes. Don't set it too big. 4-8 are good values.
        "num_val_loader_workers": 4,    // number of evaluation data loader processes.
        "batch_group_size": 0,  //Number of batches to shuffle after bucketing.
        "min_seq_len": 3,       // DATASET-RELATED: minimum text length to use in training
        "max_seq_len": 500,     // DATASET-RELATED: maximum text length
        "compute_f0": false,     // compute f0 values in data-loader
        "use_noise_augment": true,  //add a random noise to audio signal for augmentation at training .
        "compute_input_seq_cache": true,

        // PATHS
        "output_path": "trained_models/ljspeech/models",

        // PHONEMES
        "phoneme_cache_path": "trained_models/ljspeech/caches",  // phoneme computation is slow, therefore, it caches results in the given folder.
        "use_phonemes": true,           // use phonemes instead of raw characters. It is suggested for better pronounciation.
        "phoneme_language": "en-us",     // depending on your target language, pick one from  https://github.com/bootphon/phonemizer#languages

        // MULTI-SPEAKER and GST
        "use_speaker_embedding": false,     // use speaker embedding to enable multi-speaker learning.
        "use_external_speaker_embedding_file": false,
        "style_wav_for_test": null,          // path to style wav file to be used in TacotronGST inference.
        "use_gst": false,       // TACOTRON ONLY: use global style tokens

        // DATASETS
        "datasets":   // List of datasets. They all merged and they get different speaker_ids.
            [
                {
                    "name": "ljspeech",
                    "path": "/home/William/Desktop/mozilla_tts/LJSpeech-1.1/",
                    "meta_file_train": "metadata.csv",
                    "meta_file_val": null
                    // "path_for_attn": "/home/erogol/Data/LJSpeech-1.1/alignments/"
                }
            ]
    }

#### Running Glow TTS training
CUDA_VISIBLE_DEVICES="1" python TTS/bin/train_glow_tts.py --config_path TTS/tts/configs/glow_tts_ljspeech.json

Stopped training after 4k steps, as avg_align_error did not change.

**My environment:**
- **Ubuntu 20.04**:
- **PyTorch 1.7.1+cu110**:
- **Python 3.8.5**:

I have tried both v0.0.12 and v0.0.13 and the issue persisted. 
Thank you and please let me know if you need more information from me.

**Comment by david-eom** [comment]
Hi Eren, I am encountering the same issue here as well. I have tried to train the Glow-TTS model with LJSpeech dataset with version 0.0.13, the `avg_align_error` in the tensorboard output likewise stays constant 5k steps into the training.

I have also experimented with different `encoder_type`, and the problem persists. All trainings are carried out with default configs from the current repo / config for released models.

* `residual_bn_conv`
![截圖 2021-07-21 上午11 57 33](https://user-images.githubusercontent.com/66581587/126428622-34c73ad2-b394-423f-a5b3-3443b14497c5.png)
`avg_align_error`: 0.05798


* `gated_conv`
![截圖 2021-07-21 下午12 01 34](https://user-images.githubusercontent.com/66581587/126428934-689c7d82-6ffa-49dd-8477-eaccb6f3e604.png)
`avg_align_error`: 0.1014


* `rel_pos_transformer`
![截圖 2021-07-21 下午12 02 38](https://user-images.githubusercontent.com/66581587/126429023-0e00d2e9-a511-46e0-997e-eb5cecfda40c.png)
`avg_align_error`: 0.1014


* `time_depth_separable`
![截圖 2021-07-21 下午12 03 57](https://user-images.githubusercontent.com/66581587/126429135-b9674356-a1a2-4787-9fc9-503aaf87f11b.png)
`avg_align_error`: 0.1014

I do understand that this is an out-of-date version, but would still like to seek your kind advice and assistance if the issue could be resolved. Thank you in advance!

**Comment by erogol** [comment]
Why do you prefer to use the older version ?

**Comment by lihaoyangML** [comment]
Hi Eren, thanks for the reply. I am using the older version because I want to experiment fine tuning the released multi-speaker glow TTS models in the repo on my own multi-speaker dataset. However, I think the pretrained models, such as [SC-GlowTTS-Trans](https://colab.research.google.com/drive/1yyQDc-xWCqa2g-d1joW_goqbYZKaImsJ?usp=sharing) are not compatible with the newer version.

**Comment by erogol** [comment]
nope they are all comptible.

**Comment by david-eom** [comment]
Hi Eren, I switched over to the latest version earlier on (v0.1.2), and it seems like the issue persists.

* Tensorboard output:
![截圖 2021-07-22 下午5 44 25](https://user-images.githubusercontent.com/66581587/126619828-200ef399-2de3-4962-b770-ac40aaa81e5c.png)
`avg_align_error` stays constant at 0.09933.

* Training script (identical to recipes/ljspeech/glow_tts/train_glowtts.py):
```python
import os

from TTS.tts.configs import GlowTTSConfig
from TTS.tts.configs import BaseDatasetConfig
from TTS.trainer import init_training, Trainer, TrainingArgs

output_path = "/home/david_yan/coqui_ai_tts/glow_tts_models"
dataset_config = BaseDatasetConfig(name="ljspeech", meta_file_train="metadata.csv", path="/home/david_yan/mozilla_tts/TTS/tts/datasets/LJSpeech-1.1/")
config = GlowTTSConfig(
    batch_size=32,
    eval_batch_size=16,
    num_loader_workers=4,
    num_eval_loader_workers=4,
    run_eval=True,
    test_delay_epochs=-1,
    epochs=1000,
    text_cleaner="english_cleaners",
    use_phonemes=False,
    phoneme_language="en-us",
    phoneme_cache_path=os.path.join(output_path, "phoneme_cache"),
    print_step=25,
    print_eval=True,
    mixed_precision=False,
    output_path=output_path,
    datasets=[dataset_config]
)
args, config, output_path, _, c_logger, tb_logger = init_training(TrainingArgs(), config)
trainer = Trainer(args, config, output_path, c_logger, tb_logger)
trainer.fit()
```

**Comment by erogol** [comment]
BTW I just realized that the align_error does not really make sense with models using duration predictors since it actually measures how diagonal and decisive the alignment is but since the alignment is 0-1 valued for such models, it computes the same error for every step. 

So the solution to this issue is to removing align_error from these models. But for now, you can just ignore it.

**Comment by BoneGoat** [comment]
"I just realized that the align_error does not really make sense with models using duration predictors..." - Is this also true for Taco2 DCA? Because I'm experiencing the same issue where avg_align_error isn't changing at all efter each epoch.

This is a new dataset I'm using and I had some success with it using Nvidia Nemo but so far Coqui GlowTTS and Taco2 is just producing noise.

**Comment by erogol** [comment]
It'd just produce noise until the attention aligns.
